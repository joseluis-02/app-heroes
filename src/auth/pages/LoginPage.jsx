import { useNavigate } from "react-router-dom"

export const LoginPage = () => {
  
  const navigate = useNavigate();

  const onLogin = () => {
    navigate('/',{
      replace: true
    })
  }

  return (
    <div className="container">
      <h1>Login</h1>
      <hr />
      <button 
        className="btn btn-success"
        onClick={ onLogin }
      >
        Iniciar sesi√≥n
      </button>

    </div>
  )
}
