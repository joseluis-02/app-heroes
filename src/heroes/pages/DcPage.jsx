import { HeroesList } from "../components"


export const DcPage = () => {
  return (
    <>
      <HeroesList
        publisher={'DC Comics'}
      />
    </>
  )
}
