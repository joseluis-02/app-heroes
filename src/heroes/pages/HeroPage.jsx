import { useMemo } from 'react';
import { Navigate, useNavigate, useParams } from 'react-router-dom';
import { getByHeroId } from '../helpers';

export const HeroPage = () => {

  const navigator = useNavigate();
  const { id } = useParams();
  const heroe = useMemo( () => getByHeroId(id), [ id ]);
  const {alter_ego,publisher,first_appearance,superhero,characters} = heroe;

  const onNavigateBack = () => {
    navigator(-1);
  }

  if( !heroe ){
    return <Navigate to={"/marvel"} />
  }

  return (
    <div className="row mt-4">
        <div className="col-md-4">
            <img 
              src={`../assets/heroes/${id}.jpg`}
              alt={superhero}
              className="img-fluid rounded animate__animated animate__backInLeft"
            />
        </div>
        <div className="col-md-8">
          <div className="card-body p-2">
            <h5 className="card-title"><span className="fw-semibold">Super héroe: </span>{superhero}</h5>
            <p className="card-text"><span className="fw-semibold">Editora: </span>{publisher}</p>
            <p className="card-text"><span className="fw-semibold">Personajes: </span>{characters}</p>
            <p className="card-text"><span className="fw-semibold">Alterar: </span>{alter_ego}</p>
            <p className="card-text"><span className="fw-semibold">Impresión: </span><small className="text-muted">{first_appearance}</small></p>
            <button
              className="btn btn-sm btn-light"
              onClick={onNavigateBack}
            >
              volver...
            </button>
          </div>
        </div>
    </div>
    
  )
}
