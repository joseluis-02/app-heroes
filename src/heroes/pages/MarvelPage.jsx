import { HeroesList } from "../components"

export const MarvelPage = () => {
  return (
    <>
      <HeroesList 
        publisher={'Marvel Comics'}
      />
    </>
  )
}
