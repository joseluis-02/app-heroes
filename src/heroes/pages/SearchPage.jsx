import { useLocation, useNavigate } from 'react-router-dom';
import { useForm } from '../../hokks/useForm';
import queryString from 'query-string';
import { getHeroesByName } from '../helpers/getHeroesByName';
import { HeroItem } from '../components/HeroItem';

export const SearchPage = () => {

  const navigate = useNavigate();
  const location = useLocation();

  const { q = '' } = queryString.parse( location.search );

  const heroes = getHeroesByName( q );

  const { searchText,onInputChange,onResetForm } = useForm({ searchText: '' });

  const onSubmitFormSearch = (event) => {
    event.preventDefault();
    //if( searchText.trim().length <= 1) return;
    navigate(`?q=${ searchText }`);
    onResetForm();
  }


  return (
    <>
      <div className="row mt-3">
        <div className="col-sm-5">

          <form aria-label="form" onSubmit={ onSubmitFormSearch }>
            <input 
              type="search"
              className="form-control"
              placeholder="Nombre del héroe"
              name="searchText"
              value={searchText}
              onChange={ onInputChange }
              aria-label="inputSearch"
             />
             <button
              className="btn btn-outline-info mt-2"
             >
              Buscar
             </button>
          </form>

        </div>

        <div className="col-sm-7">
          <div className="shadow-none border-bottom p-3 mb-3 bg-body rounded">Resultados</div>
          {
            ( q.length === 0 )
            ?  <div className="alert alert-info animate__animated animate__fadeIn">Buscar un héroe</div>
            : ( q.length > 0 && heroes.length === 0) && <div className="alert alert-danger animate__animated animate__fadeIn">No hay héroe con el nombre: <b>{ q }</b></div>
          }
          {
            heroes.map( heroe => (
              <HeroItem 
                key={ heroe.id}
                {...heroe}
              />
            ))
          }
        </div>

      </div>
    </>
  )
}
