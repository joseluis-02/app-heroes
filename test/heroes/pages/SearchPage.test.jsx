import {fireEvent, render, screen} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { SearchPage } from '../../../src/heroes/pages/SearchPage';

const mockUseNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockUseNavigate
}));

describe('Pruebas en mi componente <SearchPage />', () => { 

    beforeEach(() => jest.clearAllMocks());

    test('Debe de mostrarse correctamente con valores por defecto', () => {
    
       const {container} = render(
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
        );

        expect( container ).toMatchSnapshot();

    });

    test('Debe de mostrar el batman y el input con el valor de queryString', () => {
        const {container} = render(
            <MemoryRouter initialEntries={['/search?q=bat']}>
                <SearchPage />
            </MemoryRouter>
        );

        //screen.debug();
        //screen.debug();
        const inputValue = screen.getByLabelText('inputSearch');
        //console.log(inputValue.value);
        expect( inputValue.value ).toBe('');

        const img = screen.getByRole('img');
        //console.log(img.src);
        expect( img.src ).toBe('http://localhost/assets/heroes/dc-batman.jpg');

        //const divDanger = screen.getByLabelText('div-danger');
    });
    
    test('Debe de mostrar un error si no se encuentra un heroe con el nombre (batman123', () => {

        const {container} = render(
            <MemoryRouter initialEntries={['/search?q=batman123']}>
                <SearchPage />
            </MemoryRouter>
        ); 

        expect( screen.getByText('batman123')).toBeTruthy();

    });

    test('Debe de llamar el navigate a la pantalla nueva ', () => {
        render(
            <MemoryRouter initialEntries={['/search']}>
                <SearchPage />
            </MemoryRouter>
        );
        
        const input = screen.getByLabelText('inputSearch');
        fireEvent.change( input, {target:{name: 'searchText', value: 'superman'}});
        const form = screen.getByLabelText('form');
        fireEvent.submit(form);
        expect( mockUseNavigate ).toHaveBeenCalledWith('?q=superman');
    });


 });