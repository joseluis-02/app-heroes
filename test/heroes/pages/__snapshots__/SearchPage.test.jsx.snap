// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en mi componente <SearchPage /> Debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <div
    class="row mt-3"
  >
    <div
      class="col-sm-5"
    >
      <form
        aria-label="form"
      >
        <input
          aria-label="inputSearch"
          class="form-control"
          name="searchText"
          placeholder="Nombre del héroe"
          type="search"
          value=""
        />
        <button
          class="btn btn-outline-info mt-2"
        >
          Buscar
        </button>
      </form>
    </div>
    <div
      class="col-sm-7"
    >
      <div
        class="shadow-none border-bottom p-3 mb-3 bg-body rounded"
      >
        Resultados
      </div>
      <div
        class="alert alert-info animate__animated animate__fadeIn"
      >
        Buscar un héroe
      </div>
    </div>
  </div>
</div>
`;
